lib/base_service.js

```diff
@@ -31,13 +32,48 @@ class BaseService {
         this._logger = null;
         this._metrics = null;
         this._ratelimiter = null;
-        this.serviceReturns = null;
+        this._serviceReturns = null;
+        this._uncaughtExceptionHandlerAdded = false;
+        this._unhandledRejectionHandlerAdded = false;
+
+        if (!this._uncaughtExceptionHandlerAdded) {
+            this._uncaughtExceptionHandlerAdded = true;
+            this._uncaughtExceptionHandler = (err) => {
+                this._logUnhandledException(err);
+                // Don't exit right away, allow logger to process message
+                setTimeout(() => {
+                    process.exit(1);
+                }, 100);
+            };
+        }
+        // Avoid recursion if there are bugs in the logging code.
+        this.inLogger = false;
```

Comment:
Let's prepend with `_`, it's a private field.

Reply:
Done